.github/workflows/compatibility-tests.yaml

```diff
@@ -0,0 +1,802 @@
+name: compatibility-tests
```

Comment:
Moved all of the "compatibility tests" over to this re-usable workflow so that we can potentially have a dual pipeline that runs Stable vs Alpha distros (AKS for example). This way we can test in a non-blocking way by making the alpha run of this workflow a non-required job.

Reply:
wouldn't it make more sense to have the CMX action mark a version as `alpha` or `stable` and then we could keep all these tests in `build-test.yaml` and then decide if a test should fail or not based on that?

Reply:
We could request that change, but IMO _we_ as the user of CMX need to be able to control what we accept to some level based on our definition of stable

Reply:
make sense - any ideas how we could make tests conditionally pass or fail if we went this route? i see `continue-on-error`, but that seems like it would still mark the job as failed?

Reply:
looking at the docs here, it seems it will allow the workflow to pass: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idcontinue-on-error